src/hotspot/share/opto/superwordVTransformBuilder.cpp

```diff
@@ -139,51 +142,72 @@ void SuperWordVTransformBuilder::build_inputs_for_scalar_vtnodes(VectorSet& vtn_
       init_req_with_scalar(n, vtn, 0);
       continue;
     } else {
-      set_all_req_with_scalars(n, vtn);
+      init_all_req_with_scalars(n, vtn);
     }
   }
 }
 
 // Create a vtnode for each pack. No in/out edges set yet.
 VTransformVectorNode* SuperWordVTransformBuilder::make_vector_vtnode_for_pack(const Node_List* pack) const {
-  uint pack_size = pack->size();
   Node* p0 = pack->at(0);
-  int opc = p0->Opcode();
-  VTransformVectorNode* vtn = nullptr;
+  const VTransformVectorNodePrototype prototype = VTransformVectorNodePrototype::make_from_pack(pack, _vloop_analyzer);
+  const int  sopc = prototype.scalar_opcode();
+  const uint vlen = prototype.vector_length();
```

Comment:
As someone that is not familiar with the Superword code: is "pack size" and "vector length" often used interchangeably? if not, then I would keep the name.

Reply:
`SuperWord` works with packs, i.e. packing scalars. So then we can measure the size of a pack, that measures how many scalars we have packed.

But `VTransform` is a "preview" of the new vectorized C2 IR. And there, we talk about the length of vectors. So I think this is exactly the right place to do the transition ;)